roles/ldap/templates/slapd.j2

```diff
@@ -0,0 +1,205 @@
+		exit 0
+	fi
+	if [ -n "$SLAPD_SENTINEL_FILE" ] && [ -e "$SLAPD_SENTINEL_FILE" ]; then
+		echo "Not starting slapd: $SLAPD_SENTINEL_FILE exists" >&2
+		exit 0
+	fi
+}
+
+# Tell the user that something went wrong and give some hints for
+# resolving the problem.
+report_failure() {
+	log_end_msg 1
+	if [ -n "$reason" ]; then
+		log_failure_msg "$reason"
+	else
+		log_failure_msg "The operation failed but no output was produced."
+
+		if [ -n "$SLAPD_OPTIONS" -o \
+		     -n "$SLAPD_SERVICES" ]; then
+			if [ -z "$SLAPD_SERVICES" ]; then
+				if [ -n "$SLAPD_OPTIONS" ]; then
+					log_failure_msg "Command line used: slapd $SLAPD_OPTIONS"
+				fi
+			else
+				log_failure_msg "Command line used: slapd -h '$SLAPD_SERVICES' $SLAPD_OPTIONS"
+			fi
+		fi
+	fi
+}
+
+# Start the slapd daemon and capture the error message if any to
+# $reason.
+start_slapd() {
+	# Make sure /var/run/slapd exists with correct permissions
+	if [ ! -d /var/run/slapd ]; then
+		mkdir -p /var/run/slapd
+		[ -z "$SLAPD_USER" ] || chown -R "$SLAPD_USER" /var/run/slapd
+		[ -z "$SLAPD_GROUP" ] || chgrp -R "$SLAPD_GROUP" /var/run/slapd
+	fi
+
+	# Make sure the pidfile directory exists with correct permissions
+	piddir=`dirname "$SLAPD_PIDFILE"`
+	if [ ! -d "$piddir" ]; then
+		mkdir -p "$piddir"
+		[ -z "$SLAPD_USER" ] || chown -R "$SLAPD_USER" "$piddir"
+		[ -z "$SLAPD_GROUP" ] || chgrp -R "$SLAPD_GROUP" "$piddir"
+	fi
+
+	if [ -z "$SLAPD_SERVICES" ]; then
+		reason="`start-stop-daemon --start --quiet --oknodo \
+			--pidfile "$SLAPD_PIDFILE" \
+			--exec $SLAPD -- $SLAPD_OPTIONS 2>&1`"
+	else
+		reason="`start-stop-daemon --start --quiet --oknodo \
+			--pidfile "$SLAPD_PIDFILE" \
+			--exec $SLAPD -- -h "$SLAPD_SERVICES" $SLAPD_OPTIONS 2>&1`"
+	fi
+
+	# Backward compatibility with OpenLDAP 2.1 client libraries.
+	if [ ! -h /var/run/ldapi ] && [ ! -e /var/run/ldapi ] ; then
+		ln -s slapd/ldapi /var/run/ldapi
+	fi
+}
+
+# Stop the slapd daemon and capture the error message (if any) to
+# $reason.
+stop_slapd() {
+	reason="`start-stop-daemon --stop --quiet --oknodo --retry TERM/10 \
+		--pidfile "$SLAPD_PIDFILE"
+# due to issues with start-stop-daemon not finding the process
+# named with --exec, we remove that option and only stop the
+# daemon based on the PID (which is risky)
+#		--exec $SLAPD 2>&1`"
+}
+
+# Start the OpenLDAP daemons
+start_ldap() {
+	trap 'report_failure' 0
+	log_daemon_msg "Starting OpenLDAP" "slapd"
+	start_slapd
+	trap "-" 0
+	log_end_msg 0
+}
+
+# Stop the OpenLDAP daemons
+stop_ldap() {
+	trap 'report_failure' 0
+	log_daemon_msg "Stopping OpenLDAP" "slapd"
+	stop_slapd
+	trap "-" 0
+	log_end_msg 0
+}
+
+case "$1" in
+  start)
+	check_for_no_start
+  	start_ldap ;;
+  stop)
+  	stop_ldap ;;
+  restart|force-reload)
+	check_for_no_start
+  	stop_ldap
+	start_ldap
+	;;
+  status)
+	status_of_proc -p $SLAPD_PIDFILE $SLAPD slapd
+	;;
+  *)
+  	echo "Usage: $0 {start|stop|restart|force-reload|status}"
+	exit 1
+	;;
+esac
```

Comment:
Do we _really_ need to include this monster?  Can't we use a simple systemd service file like this?  https://github.com/ioanrogers/systemd-units/blob/master/system/slapd.service

if the init.d file is really necessary, I would prefer using `lininfile` or `replace` andible modules (if possible), so we at least know what we're changing in upstream's init file.